src/connectedk8s/azext_connectedk8s/_precheckutils.py

```diff
@@ -0,0 +1,219 @@
+                telemetry.set_exception(exception=error_kubectl_delete_helm.decode("ascii"), fault_type=consts.Pre_Onboarding_Inspector_Failed_Fault_Type, summary="Error while executing pre onboarding inspector Job")
+                return
+        try:
+            chart_path = azext_utils.get_chart_path(consts.Pre_Onboarding_Inspector_Job_Registry_Path, kube_config, kube_context, helm_client_location, 'ConnectPrecheckCharts')
+
+            helm_install_release_preonboarding_inspector(chart_path, http_proxy, https_proxy, no_proxy, proxy_cert, kube_config, kube_context, helm_client_location)
+        # To handle the Exception that occured
+        except Exception as e:
+            logger.warning("An error occured while installing helm release of pre onboarding inspector in the cluster. Exception:")
+            logger.warning(str(e))
+            telemetry.set_exception(exception=e, fault_type=consts.Pre_Onboarding_Inspector_Helm_Install_Failed_Fault_Type, summary="Error while installing pre onboarding inspector helm release")
+            # Deleting all the stale resources that got created
+            Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+            return
+        # Watching for pre onboarding inspector container to reach in completed stage
+        w = watch.Watch()
+        is_job_complete = False
+        is_job_scheduled = False
+        # To watch for changes in the pods states till it reach completed state or exit if it takes more than 180 seconds
+        for event in w.stream(batchv1_api_instance.list_namespaced_job, namespace='default', label_selector="", timeout_seconds=60):
+            try:
+                # Checking if job get scheduled or not
+                if event["object"].metadata.name == "pre-onboarding-inspector-job":
+                    is_job_scheduled = True
+                # Checking if job reached completed stage or not
+                if event["object"].metadata.name == "pre-onboarding-inspector-job" and event["object"].status.conditions[0].type == "Complete":
+                    is_job_complete = True
+                    w.stop()
+            except Exception as e:
+                continue
+            else:
+                continue
+
+        if (is_job_scheduled is False):
+            logger.warning("Unable to schedule the pre onboarding inspector job in the kubernetes cluster. The possible reasons can be presence of a security policy or security context constraint (SCC) or it may happen becuase of lack of ResourceQuota.\n")
+            Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
+            return
+        elif (is_job_scheduled is True and is_job_complete is False):
+            logger.warning("Unable to finish the pre onboarding inspector job in the kubernetes cluster. The possible reasons can be resource constraints on the cluster.\n")
+            Popen(cmd_helm_delete, stdout=PIPE, stderr=PIPE)
```

Comment:
same comment. Collect telemetry and set user fault

Reply:
added set exception , not user fault here , we already will raise a user fault in the end in connect code when our checks fail

Reply:
I still can't see the set exception, where is it added?

Reply:
Nit: Instead of couldn't complete, let's rephrase it to - Cluster diagnostics job didn't reach completed state